include/yaml-cpp/node/detail/impl.h

```diff
@@ -106,7 +106,11 @@ inline bool node::equals(const T& rhs, shared_memory_holder pMemory) {
 }
 
 inline bool node::equals(const char* rhs, shared_memory_holder pMemory) {
-  return equals<std::string>(rhs, pMemory);
+  std::string lhs;
+  if (convert<std::string>::decode(Node(*this, std::move(pMemory)), lhs)) {
```

Comment:
`std::move(pMemory)` can optimize the performance with move semantics, is my understading right? Why not add this in line102? So that we do not need to change the line 109.

Reply:
I'd like to make `shared_memory_holder` optimizations (move semantics or simply passing it as a `const&`) as a whole in different PRs.
This change however, is not about adding `std::move`, but not creating `std::string` from `rhs` just to compare it to `lhs`. We can do it without creating temporary string.